pkg/updater/updater.go

```diff
@@ -3,254 +3,252 @@ package updater
-type lastUpdateCheck struct {
-	Date       time.Time     `yaml:"date"`
-	CheckEvery time.Duration `yaml:"checkEvery"`
-}
+// Disabled globally disables the automatic updater. This is helpful
+// for when using an external package manager, such as brew.
+// This should usually be done with an ldflag.
+var Disabled bool
 
-// userConfig us user configuration for checking for updates for a given
-// repository
-type userConfig struct {
-	// AlwaysUsePrereleases instructs the updater to always consider prereleases
-	AlwaysUsePrereleases bool `yaml:"alwaysUsePrereleases"`
-}
+// This block contains directories the updater uses
+var (
+	// cacheDir is the directory where we store the cache of releases
+	cacheDir = filepath.Join(".outreach", ".cache", "updater")
 
-// `NeedsUpdate` reads a GitHub token from `~/.outreach/github.token`. If it can't,
-// it prompts the user and saves it to the path above. GitHub Releases is checked
-// for a new release, and if it's found, updates the base binary. If an update is
-// required, it returns `true`.
-//
-// If `repo` is not provided, it is determined via `debug.ReadBuildInfo` and
-// the path the binary was built from. `repo` should be in the format of
-// `getoutreach/$repoName`.
-//
-// `version` is usually the value of `gobox/pkg/app.Version`, which is set
-// at build time.
-//
-// If `debugLog` is set to true, then the core updater's debug logging will be
-// enabled. Deprecated: This should be set on the provided logger.
-//
-// Update checks can be disabled by setting `disabled` to true.
-//nolint:funlen,gocyclo
-func NeedsUpdate(ctx context.Context, log logrus.FieldLogger, repo, version string, disabled,
-	debugLog, includePrereleases, forceCheck bool) bool {
-	if disabled {
-		return false
-	}
+	// configDir is the directory where we store the user's configuration for
+	// the updater per repository.
+	configDir = filepath.Join(".outreach", ".config")
+)
 
-	// Never update when device is not a terminal, or when in a CI environment. However,
-	// we allow forceCheck to override this.
-	if !forceCheck && (!term.IsTerminal(int(os.Stdin.Fd())) || os.Getenv("CI") != "") {
-		return false
+// UseUpdater creates an automatic updater.
+//nolint:revive // Why: Purposely not exported.
+func UseUpdater(ctx context.Context, opts ...Option) (*updater, error) {
+	u := &updater{
+		disabled: Disabled,
 	}
-	log = log.WithField("service", "updater")
 
-	if repo == "" {
-		r, err := getRepoFromBuild()
-		if err != nil {
-			log.WithError(err).Error("failed to determine which repository build this module")
-			return false
-		}
-		repo = r
+	// parse the provided options
+	for _, opt := range opts {
+		opt(u)
 	}
 
-	split := strings.Split(repo, "/")
-	org := split[0]
-	repoName := split[1]
-
-	homedir, err := os.UserHomeDir()
-	if err != nil {
-		log.WithError(err).Warn("failed to get user's home directory")
-		return false
+	if u.log == nil {
```

Comment:
this comment also applies to the conditionals that start on line 57 and 62

this pattern you're using right now is ugly and just going to continue to bloat this function if you ever need to extend the defaults. look at this pattern i use and consider it - 

https://github.com/getoutreach/tollmon/blob/main/pkg/tollgate/tollgate.go#L106-L112
https://github.com/getoutreach/tollmon/blob/main/pkg/tollgate/options.go#L127

Reply:
it might apply to line 68 too